### PR TITLE
 info::device::atomic64 is deprecated

### PR DESCRIPTION
**Pull request by dklochkov-emb**
1. A separate .def file was created for deprecated features that are not part of SYCL 2020 spec
2. info::device::atomic64 was deprecated
3. Test for  info::device::atomic64 deprecated was added into separate file due to -Wno-deprecated-declarations flag in warnings.cpp tests file.

### CODE REVIEW

**Comment by KornevNikita** [comment]
> Test for info::device::atomic64 deprecated was added into separate file due to -Wno-deprecated-declarations flag in warnings.cpp tests file.

Just curious - did you try warnings.cpp actually? Not sure, but I think I tried it sometime ago and there was no need for a separate test.

**Comment by AlexeySachkov** [comment]
> > I'd also suggest to update our [code](https://github.com/intel/llvm/blob/sycl/sycl/source/detail/device_impl.cpp#L378), so we don't use this deprecated query.
> 
> That's a good point! `warnings.cpp` should be updated to fail unless this is fixed to ensure that we also cover this use case. I suppose that not all code paths may produce warnings, simply because we don't instantiate those templates in the test.

I've realized the the call we perform happens within the library, so it doesn't affect public headers and can't be checked by `warnings.cpp`. I agree with Nikita that we should address it (most likely our `-Werror` build in post-commit will fail otherwise), but please disregard the comment about updating `warnings.cpp`

**Comment by AlexeySachkov** [comment]
@cperkinsintel, @intel/llvm-reviewers-runtime, could you please take a look?

File: sycl/test-e2e/Basic/windows_version_agnostic_sycl_lib.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // REQUIRES: windows
 
-// RUN: %clangxx --driver-mode=cl /std:c++17 /EHsc -I%sycl_include -I%opencl_include_dir %s -o %t.out /link /defaultlib:%sycl_static_libs_dir/sycl.lib
+// RUN: %clangxx -Wno-error=deprecated-declarations --driver-mode=cl /std:c++17 /EHsc -I%sycl_include -I%opencl_include_dir %s -o %t.out /link /defaultlib:%sycl_static_libs_dir/sycl.lib
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Basic/windows_version_agnostic_sycl_lib.cpp:3]: I think this is a dangerous change...

Comment (dklochkov-emb) [reply sycl/test-e2e/Basic/windows_version_agnostic_sycl_lib.cpp]: According to agreement, I changed it to `-isystem`(in this case it is `/imsvc`  due to ` %clangxx --driver-mode=cl`)

File: sycl/test-e2e/Config/kernel_from_file.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
 // RUN: %if linux %{ llvm-link -o=%t_app.bc %t.bc %t_compiler_wrappers.bc %t_sanitizer.bc %} %else %{ llvm-link -o=%t_app.bc %t.bc %t_compiler_wrappers.bc %}
 // >> ---- translate to SPIR-V
 // RUN: llvm-spirv -o %t.spv %t_app.bc
-// RUN: %clangxx -Wno-error=ignored-attributes -DSYCL_DISABLE_FALLBACK_ASSERT %cxx_std_optionc++17 %include_option %t.h %s -o %t.out %sycl_options -fno-sycl-dead-args-optimization -Xclang -verify-ignore-unexpected=note,warning
+// RUN: %clangxx -Wno-error=ignored-attributes -Wno-error=deprecated-declarations -DSYCL_DISABLE_FALLBACK_ASSERT %cxx_std_optionc++17 %include_option %t.h %s -o %t.out %sycl_options -fno-sycl-dead-args-optimization -Xclang -verify-ignore-unexpected=note,warning
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Config/kernel_from_file.cpp:17]: Why don't we change `%include_option` to `-isystem` as well?

Comment (dklochkov-emb) [reply sycl/test-e2e/Config/kernel_from_file.cpp]: Fixed

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**